src/ctnetlink/nlas/ct_attr.rs

```diff
@@ -0,0 +1,200 @@
+        let attr_type = buf.kind();
+        let value_l = (length as usize) - NLA_HEADER_SIZE;
+        let value = buf.value();
+        if is_nested {
+            let mut nested_attrs = vec![];
+            let mut read = 0;
+            while value_l > read {
+                let nla_buf = NlaBuffer::new(&value[read..]);
+                let attr = Self::parse(&nla_buf)?;
+                read += attr.length as usize;
+                if attr.length % 4 != 0 {
+                    read += 4 - (attr.length as usize % 4);
+                }
+                nested_attrs.push(attr);
+            }
+            Ok(CtAttr {
+                nested: Some(nested_attrs),
+                length,
+                attr_type,
+                value: None,
+            })
+        } else {
+            Ok(CtAttr {
+                nested: None,
+                attr_type,
+                // padding bytes are not included
+                length,
+                value: Some(value[..value_l].to_vec()),
+            })
+        }
+    }
+}
+
+impl CtAttr {
+    pub fn is_nested(&self) -> bool {
+        self.nested.is_some()
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct CtAttrBuilder {
+    nested: Option<Vec<CtAttr>>,
+    attr_type: u16,
+    value: Option<Vec<u8>>,
+    length: u16,
+}
+
+impl CtAttrBuilder {
+    pub fn new(attr_type: u16) -> CtAttrBuilder {
+        CtAttrBuilder {
+            nested: None,
+            attr_type,
+            value: None,
+            length: 0,
+        }
+    }
+    pub fn nested_attr(mut self, attr: CtAttr) -> Self {
+        self.length += attr.length;
+        if attr.length % 4 != 0 {
+            self.length += 4 - (attr.length % 4);
+        }
+        if let Some(ref mut nested) = self.nested {
+            nested.push(attr);
+        } else {
+            self.nested = Some(vec![attr]);
+        }
+        self.attr_type |= NLA_F_NESTED;
+        self
+    }
+
+    pub fn value(mut self, v: &[u8]) -> Self {
+        self.length += v.len() as u16;
+        self.value = Some(v.to_vec());
+        self
+    }
+
+    pub fn build(&self) -> CtAttr {
+        CtAttr {
+            nested: self.nested.clone(),
+            attr_type: self.attr_type,
+            length: self.length + NLA_HEADER_SIZE as u16,
+            value: self.value.clone(),
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use netlink_packet_utils::{nla::NlaBuffer, Emitable, Parseable};
+
+    use crate::{
+        constants::{
+            CTA_IP_V4_DST, CTA_IP_V4_SRC, CTA_PROTO_DST_PORT, CTA_PROTO_NUM,
+            CTA_PROTO_SRC_PORT, CTA_TUPLE_IP, CTA_TUPLE_PROTO,
+        },
+        ctnetlink::nlas::ct_attr::CtAttr,
+    };
+    const DATA: [u8; 48] = [
+        20, 0, 1, 128, 8, 0, 1, 0, 1, 2, 3, 4, 8, 0, 2, 0, 1, 2, 3, 4, 28, 0,
+        2, 128, 5, 0, 1, 0, 17, 0, 0, 0, 6, 0, 2, 0, 220, 210, 0, 0, 6, 0, 3,
+        0, 7, 108, 0, 0,
+    ];
+
+    #[test]
+    fn test_ct_attr_parse() {
```

Comment:
I am expecting test case looks like:

 https://github.com/rust-netlink/netlink-packet-route/blob/main/src/link/tests/vrf.rs#L68

(You do not need to document every bits)


You may use nlmon to capture real netlink message:
https://github.com/rust-netlink/netlink-packet-route?tab=readme-ov-file#development